src/libostree/ostree-repo.c

```diff
@@ -2219,6 +2219,19 @@ ostree_repo_open (OstreeRepo    *self,
     self->tmp_expiry_seconds = g_ascii_strtoull (tmp_expiry_seconds, NULL, 10);
   }
 
+  { g_autofree char *compression_level_str = NULL;
+
+    /* gzip defaults to 6 */
+    (void)ot_keyfile_get_value_with_default (self->config, "archive", "zlib-level", NULL,
+                                            &compression_level_str, NULL);
+
+    if (compression_level_str)
+      /* Ensure level is in [1,9] */
+      self->zlib_compression_level = MAX (1, MIN (9, g_ascii_strtoull (compression_level_str, NULL, 10)));
```

Comment:
Eh...it's really annoying to do, we don't do it in various other places either.  It'd probably be worth adding a validating version of the API to GLib though.